workbench-core/accounts/src/services/projectService.ts

```diff
@@ -254,6 +255,55 @@ export default class ProjectService {
     return newProject;
   }
 
+  /**
+   * Update the name or description of an existing project.
+   *
+   * @param request - a {@link UpdateProjectRequest} object that contains the id of the project to update
+   *                  as well as the new field values
+   * @returns a {@link Project} object that reflects the changes requested
+   */
+  public async updateProject(request: UpdateProjectRequest): Promise<Project> {
+    const { projectId, updatedValues } = request;
+
+    // disregard empty strings as valid input
+    if (updatedValues.name !== undefined && _.isEmpty(updatedValues.name)) {
```

Comment:
We might want these validation checks in the parser because the parser shouldn't be outputting invalid request object. What do you think?

Reply:
Going to try using zod [.refine](https://zod.dev/?id=refine) for this